### PR TITLE
Updates to dependencies and GitHub workflows

### PR DESCRIPTION
**Pull request by gaby**
- Bump all dependencies
- Add dependabot to track go and github-action dependencies automatically
- Update the github-workflow to test `gotestsum` against the latest Golang version and multiple platforms

Note: The following tests are failing:
```
=== RUN   TestE2E_RerunFails/first_run_has_errors,_abort_rerun
    main_e2e_test.go:69: assertion failed: 
        --- expected
        +++ actual
        @@ -1,6 +1,11 @@
        +FAIL testjson/internal/broken
        +
        +=== Failed
        +=== FAIL: testjson/internal/broken 
        +FAIL	gotest.tools/gotestsum/testjson/internal/broken [build failed]
         
         === Errors
         ../testjson/internal/broken/broken.go:5:21: undefined: somepackage
         
        -DONE 0 tests, 1 error
        +DONE 0 tests, 1 failure, 1 error
         
        
        
        You can run 'go test . -update' to automatically update testdata/e2e/expected/TestE2E_RerunFails/first_run_has_errors,_abort_rerun to the new expected value.'
        
--- FAIL: TestE2E_RerunFails/first_run_has_errors,_abort_rerun (0.06s)
FAIL cmd.TestE2E_RerunFails/first_run_has_errors,_abort_rerun (0.06s)
=== RUN   TestE2E_RerunFails
--- FAIL: TestE2E_RerunFails (1.66s)
FAIL cmd.TestE2E_RerunFails (1.66s)
```

```
FAIL cmd

=== Skipped
=== SKIP: testjson TestNewDotFormatter (0.00s)
WARN Failed to detect terminal width for dots format, error: inappropriate ioctl for device
    dotformat_test.go:135: !ok: no terminal width

=== SKIP: testjson TestGetPkgPathPrefix/with_go_path (0.00s)
    pkgpathprefix_test.go:22: isGoModuleEnabled()

=== Failed
=== FAIL: cmd TestE2E_RerunFails/first_run_has_errors,_abort_rerun (0.06s)
    main_e2e_test.go:69: assertion failed: 
        --- expected
        +++ actual
        @@ -1,6 +1,11 @@
        +FAIL testjson/internal/broken
        +
        +=== Failed
        +=== FAIL: testjson/internal/broken 
        +FAIL	gotest.tools/gotestsum/testjson/internal/broken [build failed]
         
         === Errors
         ../testjson/internal/broken/broken.go:5:21: undefined: somepackage
         
        -DONE 0 tests, 1 error
        +DONE 0 tests, 1 failure, 1 error
         
        
        
        You can run 'go test . -update' to automatically update testdata/e2e/expected/TestE2E_RerunFails/first_run_has_errors,_abort_rerun to the new expected value.'
        
=== FAIL: cmd TestE2E_RerunFails (1.66s)
```

### CODE REVIEW

**Comment by gaby** [comment]
@dnephin Can you take a look at this PR

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -5,13 +5,18 @@ on:
   push:
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  Build:
+    strategy:
+      matrix:
+        go-version: [stable, oldstable]
```
Comment (gaby) [inline .github/workflows/ci.yaml:11]: Should this run go1.20 too?

Comment (dnephin) [reply .github/workflows/ci.yaml:11]: This seems like an easier way to manage the versions. I think it's fine if 1.20 only runs on circleci. Maybe it doesn't even need to run there.

**Review by dnephin — APPROVED** [review]
Thank you for the PR!

I think I fixed the test failures, and adjusted the config slightly.

**Comment by dnephin** [comment]
Ah, the `go/version` package is too new as well. We need a separate file for the import.

**Comment by gaby** [comment]
> Ah, the `go/version` package is too new as well. We need a separate file for the import.

Yeah, go/version is only available in 1.22 or higher.

https://pkg.go.dev/go/version?tab=versions

**Comment by gaby** [comment]
All jobs are green! 💪

**Comment by dnephin** [comment]
Thank you again for the PR!

**Comment by gaby** [comment]
> Thank you again for the PR!

Thanks for the awesome tool! 💪

**Pull request merged**